### PR TITLE
Better handling of syntax errors 

### PR DESCRIPTION
**Pull request by jerelmiller**
When encountering a error while parsing the source file for GraphQL query strings (such as a syntax error), it was difficult to determine which file caused the error.

This PR adds a couple additional improvements:
- Reports the error in a similar way as other errors, showing the filename and location of the error
- Allows syntax errors to be reported alongside other errors, rather than halting the program on the first syntax error. This means that if multiple files have syntax errors, all of them will be displayed in the error output.

**Before**

```
$ npx generate-persisted-query-manifest
/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:64
}, descriptor)), Object.assign(new constructor(), properties));
                               ^

SyntaxError: Missing initializer in const declaration. (3:11)
    at instantiate (/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:64:32)
    at constructor (/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:361:12)
    at JSXParserMixin.raise (/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:3251:19)
    at JSXParserMixin.parseVar (/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:13186:16)
    at JSXParserMixin.parseVarStatement (/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:13017:10)
    at JSXParserMixin.parseStatementContent (/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:12621:23)
    at JSXParserMixin.parseStatementLike (/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:12535:17)
    at JSXParserMixin.parseModuleItem (/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:12519:17)
    at JSXParserMixin.parseBlockOrModuleBlockBody (/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:13106:36)
    at JSXParserMixin.parseBlockBody (/Users/jerelmiller/code/apollo-utils/node_modules/@babel/parser/lib/index.js:13099:10) {
  code: 'BABEL_PARSER_SYNTAX_ERROR',
  reasonCode: 'DeclarationMissingInitializer',
  loc: Position { line: 3, column: 11, index: 50 },
  pos: [Getter/Setter]
}

Node.js v20.4.0
```

**After**
```
$ npx generate-persisted-query-manifest
src/bad.graphql
   3:1  error  GraphQLError: Syntax Error: Expected Name, found <EOF>.

src/test.js
  3:11  error  SyntaxError: Missing initializer in const declaration. (3:11)

✖ 2 errors
```

### CODE REVIEW

File: packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts [hunk 1]
```diff
@@ -1034,7 +1096,19 @@ test("gathers and reports all errors together", async () => {
       "1:1  error  Operation named "HelloQuery" already defined in: src/hello-query.graphql",
       "src/query.graphql",
       "1:1  error  Anonymous GraphQL operations are not supported. Please name your query.",
-      "✖ 5 errors",
+      "src/syntax-error.graphql",
+      "2:11  error  GraphQLError: Syntax Error: Expected Name, found <EOF>.",
+      "at syntaxError (node_modules/graphql/error/syntaxError.js:15:10)",
```
Comment (glasser) [inline packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts:1101]: I personally subscribe to the philosophy that when CLI tools report errors that result from bad user input, a stack trace (in the CLI's own code) shouldn't be printed. (Only if the tool is printing out an error that indicates an unexpected bug in the CLI tool is an internal stack trace helpful.) What do you think?

Comment (jerelmiller) [reply packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts:1101]: That is totally reasonable! My only reasoning is that it made it that much more obvious that there was something horribly wrong with this file, but not sure its a good enough justification to keep it.

Here is what it looks like removed:

```
$ npx generate-persisted-query-manifest
src/bad.graphql
   3:1  error  GraphQLError: Syntax Error: Expected Name, found <EOF>.

src/test.js
  3:11  error  SyntaxError: Missing initializer in const declaration. (3:11)

✖ 2 errors
```

Better?

Comment (jerelmiller) [reply packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts]: [31c7076](https://github.com/apollographql/apollo-utils/pull/345/commits/31c707608639caf94a137fcc35253e4b4f1ca0c6)

Comment (glasser) [reply packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts]: I do think that's better — it doesn't seem like understanding the internal details of Babel etc is helpful for our users. (If we find that folks run into problems that do actually involve debugging babel, we can add the trace back in some sort of debug mode?)

**Review by glasser — APPROVED** [review]

File: packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts (hunk 1 above)
Comment (jerelmiller) [reply packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts]: Yep I like that idea! Let's leave it off for now and can revisit if/when we need it later.

**Pull request merged**